modules/ROOT/pages/batch-reference.adoc

```diff
@@ -0,0 +1,193 @@
+= Batch Component Reference
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+Mule batch components are designed to manage bulk processing of records with minimal adjustments to default component settings. However, batch components are highly configurable to meet specific processing and performance requirements.
+
+A Batch Job component must contain at least one Batch Step component. Optionally, each Batch Step component can contain a single Batch Aggregator component. The Mule flow in the following figure shows batch components.
+
+image::batch-flow-generic-example.png[alt=Batch Processing Components in a Mule Flow within Studio]
+
+The Batch Job component splits the payload of the Mule message into records. Within the Batch Step and Batch Aggregator components, these records are available with the keyword `payload`, and Mule variables are also accessible through the `vars` keyword. However, Mule attributes are not accessible from within these components. Both components return `null` on `attributes`.
+
+This reference assumes that you are familiar with the xref:batch-processing-concept.adoc#batch-job-processing-phases[batch processing phases] of a batch job instance.
+
+[[ref_batch_job_component]]
+== Batch Job Component (<batch:job />)
+
+This component prepares its input for processing and outputs a report with results of that processing. Preparation takes place during the xref:batch-processing-concept.adoc#load-and-dispatch[Load and Dispatch] phase of a batch job instance, and the report is available in the xref:batch-processing-concept.adoc#on-complete[On Complete] phase.
+
+In Studio, you can set these properties in the component's <<ref_batch_job_general_fields, General>> and <<ref_batch_job_history_fields, History>> tabs.
+
+[[ref_batch_job_general_fields]]
+=== Batch Job Properties
+
+The Batch Job component provides a number of configurable properties that determine how batch processing takes place. It also provides a way of storing the input payload in target variable for use by subsequent components in the flow, which can be useful because the Batch Job component consumes the message payload.
```

Comment:
```suggestion
The Batch Job component provides a number of configurable properties that determine how batch processing takes place. It also provides a way of storing the input payload in a target variable for use by subsequent components in the flow, which can be useful because the Batch Job component consumes the message payload.
```